### PR TITLE
in _axes.py: Added repr to error messages instead of using str

### PR DESCRIPTION
**Pull request by Cloverwave**
## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by jklymak** [comment]
Can you provide some context as to why you are doing this?   Or maybe mark as draft until you are ready?

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -834,8 +834,8 @@ def _check_no_units(vals, names):
         # Helper method to check that vals are not unitized
         for val, name in zip(vals, names):
             if not munits._is_natively_supported(val):
-                raise ValueError(f"{name} must be a single scalar value, "
-                                 f"but got {val}")
+                raise ValueError(f"{repr(name)} must be a single scalar value,"
+                                 f" but got {repr(val)}")
```
Comment (oscargus) [inline lib/matplotlib/axes/_axes.py:838]:
```suggestion
                                 f" but got {val!r}")
```
and so on also works (and is more compact).

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -2402,9 +2402,10 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
                 try:
                     left = x - width / 2
                 except TypeError as e:
-                    raise TypeError(f'the dtypes of parameters x ({x.dtype}) '
-                                    f'and width ({width.dtype}) '
-                                    f'are incompatible') from e
+                    raise TypeError('the dtypes of parameters x'
+                                    f' ({repr(x.dtype)}) '
```
Comment (oscargus) [inline lib/matplotlib/axes/_axes.py:2406]: I'm not sure that we want repr here.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -2899,7 +2900,7 @@ def stem(self, *args, linefmt=None, markerfmt=None, basefmt=None, bottom=0,
         """
         if not 1 <= len(args) <= 5:
             raise TypeError('stem expected between 1 and 5 positional '
-                            'arguments, got {}'.format(args))
+                            f'arguments, got {repr(args)}')
```
Comment (oscargus) [inline lib/matplotlib/axes/_axes.py:2903]: This will most likely not produce the expected results.

File: lib/matplotlib/axes/_axes.py [hunk 4]
```diff
@@ -3124,7 +3125,8 @@ def get_next_color():
 
         _api.check_isinstance(Number, radius=radius, startangle=startangle)
         if radius <= 0:
-            raise ValueError(f'radius must be a positive number, not {radius}')
+            raise ValueError('radius must be a positive number, not'
+                            f' {repr(radius)}')
```
Comment (oscargus) [inline lib/matplotlib/axes/_axes.py:3129]: This doesn't make sense.

**Comment by oscargus** [comment]
I take it that this is related to https://github.com/matplotlib/matplotlib/issues/21959

However, we do not *always* want the repr. Typically, for string input it makes sense (as one a) do not have to type the "/' and b) any line breaks or special characters are shown more clearly). But for e.g. a float value or a type it will not make any difference, it will print the same and is just confusing to have the additional call.

**Comment by anntzer** [comment]
Agreed with @oscargus; also note that this could/should be spelled `{foo!r}` instead of `{repr(foo)}`, for conciseness...

**Comment by jklymak** [comment]
I'm moving to draft until revised.  Thanks!

**Pull request closed without merging**